### PR TITLE
[GPU] MinGW build fixes:

### PR DESCRIPTION
**Pull request by sezero**
This fixes build against mingw-w64 version 8.0.3:

- CreateEventEx() is guarded by _WIN32_WINNT >= 0x0600:
  In SDL_gpu_d3d12.c, include core/windows/SDL_windows.h
  that defines _WIN32_WINNT properly to account for that
- DXGIInfoQueue stuff is not available in all toolchains
  such as mingw-w64 version 8.0.3
- SDL_gpu_d3d11.c: _WIN32 is always defined by MinGW, as
  is the case for all windows-targeting compilers, so it
  doesn't guarantee absence of DXGIInfoQueue stuff:
  rely on __IDXGIInfoQueue_INTERFACE_DEFINED__, as it is
  done elsewhere.

Fixes: https://github.com/libsdl-org/SDL/issues/10705 .

### CODE REVIEW

**Comment by sezero** [comment]
UWP builder fails like this:
```
D:\a\SDL\SDL\src\gpu\d3d12\../../video/directx/SDL_d3d12.h(33): error C2220: the following warning is treated as an error
D:\a\SDL\SDL\src\gpu\d3d12\../../video/directx/SDL_d3d12.h(33): warning C4005: '__REQUIRED_RPCNDR_H_VERSION__': macro redefinition
C:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\shared\wtypes.h(13): note: see previous definition of '__REQUIRED_RPCNDR_H_VERSION__'
```

I guess SDL_d3d12.h relies on windows.h not being included before it..
If it can't be worked around, I can `#define _WIN32_WINNT 0xA00` at top of
SDL_gpu_d3d12.c instead of including SDL_windows.h.. Comments?

**Comment by sezero** [comment]
> I guess SDL_d3d12.h relies on windows.h not being included before it.. If it can't be worked around, I can `#define _WIN32_WINNT 0xA00` at top of SDL_gpu_d3d12.c instead of including SDL_windows.h.. Comments?

I tried doing that locally, but hit another thing instead:

```
/tmp/SDL3/src/gpu/d3d12/SDL_gpu_d3d12.c:26:0: warning: "_WIN32_WINNT" redefined
 #define _WIN32_WINNT 0xA00 //#include "../../core/windows/SDL_windows.h"
 
In file included from /opt/cross_win64/x86_64-w64-mingw32/include/corecrt.h:10:0,
                 from /opt/cross_win64/x86_64-w64-mingw32/include/crtdefs.h:10,
                 from /opt/cross_win64/x86_64-w64-mingw32/include/sys/types.h:13,
                 from /tmp/SDL3/src/SDL_internal.h:84,
                 from /tmp/SDL3/build/CMakeFiles/SDL3-shared.dir/cmake_pch.h:4,
                 from <command-line>:0:
/opt/cross_win64/x86_64-w64-mingw32/include/_mingw.h:233:0: note: this is the location of the previous definition
 #define _WIN32_WINNT 0x502
```

I guess SDL_d3d12.h needs some adjustment?? CC: @flibitijibibo

**Comment by flibitijibibo** [comment]
Odd, I thought we were already including it in there for stuff like winapifamily and the rpcndr fix - I'm not able to check this out until next week but I let the rest of the team know. The Windows path is neatly separated from Xbox so whatever makes CI happy will be safe to apply.

**Comment by sezero** [comment]
Well, at this point, I'm waiting for ideas

**Comment by thatcosmonaut** [comment]
Build system issues are really not my strong suit unfortunately. We might have to wait for Ethan to look into this.

**Comment by sezero** [comment]
BTW, is UWP a D3D12 target ?

**Comment by thatcosmonaut** [comment]
No, UWP is completely deprecated on Xbox and UWP submissions are no longer allowed.

**Comment by sezero** [comment]
> No, UWP is completely deprecated on Xbox and UWP submissions are no longer allowed.

So, why are we building d3d11 and d3d12 gpu code for uwp ?

**Comment by thatcosmonaut** [comment]
I guess a bigger question might be why SDL3 supports UWP, given that it has been deprecated for almost 4 years now and Microsoft has abandoned it on all relevant platforms.

**Comment by flibitijibibo** [comment]
Was about to say the same thing - I removed it as part of #6570 but it got added back soon after; AFAIK _no_ UWP submissions are accepted anywhere as of writing.

**Comment by thatcosmonaut** [comment]
@slouken Can we reconsider UWP support? UWP submissions are no longer allowed on any platform and the bitrotting toolchain has caused us more than a few issues at this point.

**Comment by slouken** [comment]
> @slouken Can we reconsider UWP support? UWP submissions are no longer allowed on any platform and the bitrotting toolchain has caused us more than a few issues at this point.

I think maybe we should remove it at this point. @icculus, thoughts?

Here was my comment from back then:
https://github.com/libsdl-org/SDL/issues/6570#issuecomment-1332740245

**Comment by sezero** [comment]
If nothing else, gpu code can be disabled for uwp target (whoever's gonna do that), this patch can go in, and uwp removal can be discussed further.

**Comment by thatcosmonaut** [comment]
I'm totally fine with that, I would do it myself but I'm not sure how.

**Comment by slouken** [comment]
> I'm totally fine with that, I would do it myself but I'm not sure how.

Sure, I can do that.

**Comment by sezero** [comment]
Testing a patch in https://github.com/sezero/SDL/tree/gpu-mingw2 -- if it passes CI build, will add here.

**Comment by sezero** [comment]
OK, looks like uwp/winrt is no longer in the way as of 0f38c1a46a3ed2ebfc42a31130b6ece98ab50c90

Can you guys review? OK to merge?

**Review by flibitijibibo — APPROVED** [review]
Latest lgtm, I'm happy if CI's happy!

**Review by slouken — APPROVED** [review]
Looks good!

**Comment by sezero** [comment]
OK, the patches are in.

You guys should create a new ticket for uwp/winrt removal, if still considered.

**Comment by Sergio0694** [comment]
> "UWP submissions are no longer allowed on any platform."

@thatcosmonaut just out it curiosity, are you able to cite what source you used for that completely incorrect statement? I'm just trying to understand whether you just misunderstood, or whether there's some inaccurate docs out there that should be fixed. Thanks!

**Comment by flibitijibibo** [comment]
Here's the Xbox part, I'll let someone else figure out the other storefronts (though I believe Win32 is accepted on all applicable storefronts, Xbox was the exception):

https://learn.microsoft.com/en-us/windows/uwp/gaming/getting-started

**Comment by Sergio0694** [comment]
That is a bit unclear, but "UWP based games are no longer accepted in the Xbox Store" refers to "UWP GDK games are no longer accepted in the Xbox Store". You can still build and publish UWP apps **and games** on the Microsoft Store, which you can make available on all supported device families (which **includes Xbox**). That's all completely supported.

I still am not sure how folks went from that quote though to claiming "this must mean it's no longer allowed on any platform". Unless there's some other quote I'm missing. But even if that were the case, the fact is that statement is still just incorrect.

**Comment by flibitijibibo** [comment]
Understood - at this stage we've migrated SDL to GDKX and the Win32 support can be used on the Microsoft Store, so our bases should be covered unless they're planning on bringing back Windows Phone any time soon.

At least from FNA's point of view as the former maintainers of the UWP support we are _really_ happy with the newer targeting options, and we're more than happy to recommend the MS Store (particularly for Smart Delivery) without needing to use the no-longer-developed WinRT stuff. I don't think dropping UWP will negatively affect the store (if anything, investing in UWP further might make it _worse_ since it's old enough to have started [bitrotting in some places](https://github.com/FNA-XNA/FNA3D/commit/b490a9ba2b866a2e43fb58866b91e66715bdc13c)), so I wouldn't lose too much sleep over it!

**Pull request merged**
